internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -127,8 +132,11 @@ func (connector *SingleUseConnector) Connect(
 	// it from the client and throws it away, so that client can advance to the next
 	// stage of the process.  Otherwise the client would block forever waiting for its
 	// pre-login handshake to be read.
-	clientBuffer := mssql.NewTdsBuffer(bufferSize, connector.clientConn)
-	_, err := mssql.ReadPreloginWithPacketType(clientBuffer, mssql.PackPrelogin)
+	connector.clientBuff = mssql.NewTdsBuffer(bufferSize, connector.clientConn)
+	_, err := mssql.ReadPreloginRequest(connector.clientBuff)
+	if err == io.EOF {
+		return nil, err
```

Comment:
Do we need to log anything here indicating an EOF was seen, maybe at info level? Otherwise, it seems the connection attempt is dropped silently.

Reply:
I think that's something that can take place in the Secretless "framework" code. I think an error of `io.EOF` from any service connector should generally mean the same thing.